src/chef_ci_status.rb

```diff
@@ -194,19 +335,152 @@ def print_pr_or_issue_stats(stats, item)
   puts "    Avg Time to Close #{item_plural}: #{avg_time_str}"
 end
 
-options = {
-  org: 'chef',
-  repo: 'chef',
-  branches: ['main'],
-  days: 30,
-  verbose: false,
-  mode: ['all'],
-}
+# Define defaults from config or use hardcoded values
+options = if defined?(Settings)
+            default_mode = if Settings.default_mode.is_a?(Array)
+                             Settings.default_mode
+                           else
+                             [Settings.default_mode]
+                           end
+            # Get CI timeout from config with fallback to default
+            # Get timeout with fallback
+            has_timeout = Settings.respond_to?(:ci_timeout)
+            ci_timeout = has_timeout ? Settings.ci_timeout : 180
+            {
+              org: Settings.default_org,
+              repo: Settings.default_repo,
+              branches: Settings.default_branches,
+              days: Settings.default_days,
+              verbose: false,
+              mode: default_mode,
+              ci_timeout:, # From config or default: 3 minutes
+            }
+          else
+            {
+              org: 'chef',
+              repo: 'chef',
+              branches: ['main'],
+              days: 30,
+              verbose: false,
+              mode: ['all'],
+              ci_timeout: 180, # Default: 3 minutes
+            }
+          end
 
 valid_modes = %w{ci pr issue all}
 OptionParser.new do |opts|
   opts.banner = 'Usage: chef_ci_status.rb [options]'
 
+  opts.on(
+    '--config CONFIG_FILE',
+    'Path to custom configuration file',
+  ) do |v|
```

Comment:
the order in which this will run is a bit unclear to me, but I think ti's possible that you could load a config from a command-line option, *THEN* parse the config here, and overwrite a command-line option, which would be bad UI.

Probably we want to:
* run OptionParser, loading the CLI config into `options`, including a config file path if one exists
* if a config-file path exists, load that into `config_file_options`
* do effectivley: `options = config_file_options.merge(options)`